framework/src/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -376,63 +384,68 @@ private void parse(TypeDeclaration typeDecl,
-                                              elt, elt.getClass(), type, type.getClass(), typeArguments.size(), typeArguments));
-                 */
+                                          decl, typeParameters,
+                                          elt, elt.getClass(), type, type.getClass(), typeArguments.size(), typeArguments));
+             */
+        }
+
+        if ((typeParameters != null) && (typeParameters.size() != typeArguments.size())) {
+            // TODO: decide how severe this problem really is; see comment above.
+            // System.out.printf("Dying.  theCompilationUnit=%s%n", theCompilationUnit);
+            if (debugStubParser) {
+                stubDebug(String.format("parseType:  mismatched sizes for params and args%n  decl=%s%n  typeParameters (size %d)=%s%n  elt=%s (%s)%n  type=%s (%s)%n  typeArguments (size %d)=%s%n  theCompilationUnit=%s%nEnd of message for parseType:  mismatched sizes for params and args%n",
+                        decl, typeParameters.size(), typeParameters,
+                        elt, elt.getClass(), type, type.getClass(), typeArguments.size(), typeArguments,
+                        theCompilationUnit));
             }
-            if ((typeParameters != null) && (typeParameters.size() != typeArguments.size())) {
-                // TODO: decide how severe this problem really is; see comment above.
-                // System.out.printf("Dying.  theCompilationUnit=%s%n", theCompilationUnit);
-                if (debugStubParser) {
-                    stubDebug(String.format("parseType:  mismatched sizes for params and args%n  decl=%s%n  typeParameters (size %d)=%s%n  elt=%s (%s)%n  type=%s (%s)%n  typeArguments (size %d)=%s%n  theCompilationUnit=%s%nEnd of message for parseType:  mismatched sizes for params and args%n",
-                                              decl, typeParameters.size(), typeParameters,
-                                              elt, elt.getClass(), type, type.getClass(), typeArguments.size(), typeArguments,
-                                              theCompilationUnit));
-                }
-                /*
+            /*
                 throw new Error(String.format("parseType:  mismatched sizes for params and args%n  decl=%s%n  typeParameters (size %d)=%s%n  elt=%s (%s)%n  type=%s (%s)%n  typeArguments (size %d)=%s%n",
-                                              decl, typeParameters.size(), typeParameters,
-                                              elt, elt.getClass(), type, type.getClass(), typeArguments.size(), typeArguments));
-                */
-            }
+                                          decl, typeParameters.size(), typeParameters,
+                                          elt, elt.getClass(), type, type.getClass(), typeArguments.size(), typeArguments));
+             */
         }
+
         annotateParameters(type.getTypeArguments(), decl.getTypeParameters());
```

Comment:
Now that this line is in the scope of local variable typeArguments, should it use it instead of `type.getTypeArguments()`?  If not, then we should reinstate the scoping.